### ISSUE TITLE
Inconsistent handling of REDIS environment variables across streaming and puma images

### ISSUE THREAD

**Issue by CSDUMMI** [mastodon/mastodon#29770]
### Steps to reproduce the problem

It seems that the redis configuration is being parsed differently in `streaming/index.js` to the rest of Mastodon. 

Specifically in `redisConfigFromEnv`:

``` 
const redisConfigFromEnv = (env) => {
  ...
  let redisPort = parseIntFromEnv(env.REDIS_PORT, 6379, 'REDIS_PORT');
  let redisDatabase = parseIntFromEnv(env.REDIS_DB, 0, 'REDIS_DB');

  /** @type {import('ioredis').RedisOptions} */
  const redisParams = {
    host: env.REDIS_HOST || '127.0.0.1',
    port: redisPort,
    db: redisDatabase,
    password: env.REDIS_PASSWORD || undefined,
  };

  return {
    redisParams,
    redisPrefix,
    redisUrl: typeof env.REDIS_URL === 'string' ? env.REDIS_URL : undefined,
  };
```
This code will cause an exception when `REDIS_PORT` contains any `NaN` string other than `""`, even when `REDIS_URL` is used to overwrite this very setting. 

This is inconsistent with how the rest of mastodon handles the redis configuration, where all other `REDIS_` env variables are ignored as long as a `REDIS_URL` is provided.

### Expected behaviour

Both streaming, sidekiq and puma should interpret the config the same

### Actual behaviour

streaming processes `REDIS_PORT` and `REDIS_DB` even when `REDIS_URL` is set.

### Technical details

If this is happening on your own Mastodon server, please fill out those:
- Ruby version: (from `ruby --version`, eg. v3.1.2)
- Node.js version: (from `node --version`, eg. v18.16.0)

**Comment by CSDUMMI** [comment]
I found that this issue is caused by running Mastodon in k8s. 

k8s adds environment variables with the name `<service name>_PORT` containing a URL to [linked services](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/envvars/envvars.go#L72). When one of those services is called `REDIS`, this invariably adds a `REDIS_PORT` environment  variable, which causes `parseIntFromEnv` to throw an error and the entire streaming service to fail.

This is behavior in compliance with the [legacy network links](https://docs.docker.com/network/links/#environment-variables) of docker and will [probably not be changed upstream](https://github.com/kubernetes/kubernetes/issues/65130).

To continue to run Mastodon in k8s, #29894 needs to be merged.

**Comment by ThisIsMissEm** [comment]
Oh, sorry, I missed this! Yeah, we probably need to rewrite `redisConfigFromEnv` completely. I see @renchap already took a look at your PR, and I agree it's not quite the right approach.

I'll try to put something together after my meetings today.

**Comment by ThisIsMissEm** [comment]
Before I can land a fix for this, I'd want to have both https://github.com/mastodon/mastodon/pull/31566 and https://github.com/mastodon/mastodon/pull/31567 land, as to not have the changes required for this to conflict with [other ongoing work](https://github.com/mastodon/mastodon/pull/30979)

**Comment by ThisIsMissEm** [comment]
We probably want to be consistent with https://github.com/mastodon/mastodon/blob/d5f02adad716520b6b9014553bc730dcef5b2f50/lib/mastodon/redis_config.rb

**Comment by ThisIsMissEm** [comment]
Okay, there's a fix for this in #31623
